src/main/java/org/openlmis/stockmanagement/repository/mapper/StockCardMapper.java

```diff
@@ -14,7 +14,7 @@
   @Select("SELECT *" +
       " FROM stock_cards" +
       " WHERE facilityid = #{facilityId}" +
-      "   AND productid = #{productId}")
+      "   AND productid = (SELECT id FROM products WHERE code = #{productCode})")
```

Comment:
I would have the query do a join with the products table using productId, and then have products.code in the where clause.